.github/workflows/test-spyre.yml

```diff
@@ -0,0 +1,29 @@
+name: test-sypre
+
+on: pull_request
+
+jobs:
+  test-spyre:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+    - name: Build docker image
+      run: docker build . -t vllm-spyre -f Dockerfile.spyre
+    - name: Run Spyre tests within docker container
+      run: |
+        docker run -i --rm --entrypoint /bin/bash vllm-spyre -c '''
+          pip install pytest sentence-transformers && \
+          python3.12 -c "from transformers import pipeline; pipeline(\"text-generation\", model=\"JackFram/llama-160m\")" && \
+          export VARIANT=$(ls /root/.cache/huggingface/hub/models--JackFram--llama-160m/snapshots/) && \
+          mkdir -p /models && \
+          ln -s /root/.cache/huggingface/hub/models--JackFram--llama-160m/snapshots/${VARIANT} /models/llama-194m && \
+          python3.12 -c "from sentence_transformers import SentenceTransformer; SentenceTransformer(\"sentence-transformers/all-roberta-large-v1\")" && \
+          export VARIANT=$(ls /root/.cache/huggingface/hub/models--sentence-transformers--all-roberta-large-v1/snapshots/) && \
+          ln -s /root/.cache/huggingface/hub/models--sentence-transformers--all-roberta-large-v1/snapshots/${VARIANT} /models/all-roberta-large-v1 && \
+          export MASTER_PORT=12355 && \
+          export MASTER_ADDR=localhost && \
+          export DISTRIBUTED_STRATEGY_IGNORE_MODULES=WordEmbedding && \
+          pip install timm && \
```

Comment:
What is `timm`?

Reply:
This is a library that comes from `requirements-tests.in` from vllm, I noticed that sometimes it allows a better tracing of the error. I added it temporarily to see the error because it didn't appear locally.

Reply:
Ah actually they they fail also in my docker container when I run all the tests with a single command, but pass when I run individually `python3.12 -m pytest tests/test_spyre_tensor_parallel.py`